### DISCUSSION TITLE
Use of Docker to create mkdocs folder and conflict with github pages "master/docs"

### DISCUSSION THREAD

**Discussion by mdeliw** [squidfunk/mkdocs-material#2377] [Q&A] [answered]
I have two questions:

1. My org offers "master/docs" as publishing source for github pages. So I am assuming that my markdowns should be in a folder other than "docs" (say "src"), and my site output should be under "docs". Is this understanding true? 

So per your documentation, the docker command to setup a folder to be mkdocs project:
`docker` run --rm -it -v ${PWD}:/docs squidfunk/mkdocs-material new .` 
would create "docs" sub-folder under root to contain the sample index.md. 

If I force this command to instead mount to "src" - that doesn't work, and I assumed this would create sample index.md under a "src" sub-folder.
`docker` run --rm -it -v ${PWD}:/src squidfunk/mkdocs-material new .` 

So my understanding is wrong. 

2. If I change the mkdocs.yml theme setting for docs_dir to point to "src" and site_dir to point "docs" - that should provide me the desired outcome right? 

Given that the docker command only allows "docs" as the markdown folder, I have to manually fix the project structure.

**Comment by squidfunk** [comment, marked as answer]
No, `/docs` is just the name of the internal mount point. You can mount anything you like, but the current working directory seems to be the best option.

**Reply by mdeliw** [reply]
Thanks for the update. Either I am using the docker command mentioned in the reference incorrectly, or it is not working as expected. If `/docs` as internal mount point can be anything, then the following *doesn't work*: 

`docker run --rm -it -v ${PWD}:/src squidfunk/mkdocs-material new .` 

<img width="813" alt="Screen Shot 2021-03-02 at 7 36 32 AM" src="https://user-images.githubusercontent.com/46644001/109664106-0b199180-7b2a-11eb-83cc-05b853d922f5.png">

Now if I run the above with `/docs` and on the scaffolded project, rename `/docs` to `/src`, adjust the mkdocs.yml settings for `docs_dir` and `site_dir`:
<img width="814" alt="Screen Shot 2021-03-02 at 7 40 09 AM" src="https://user-images.githubusercontent.com/46644001/109664657-985ce600-7b2a-11eb-8a3b-bdbc996437df.png">

then the following doesn't work either: 
`docker run --rm -it -v ${PWD}:/src squidfunk/mkdocs-material build`
<img width="806" alt="Screen Shot 2021-03-02 at 7 40 33 AM" src="https://user-images.githubusercontent.com/46644001/109664776-be828600-7b2a-11eb-9aa1-dea54fb722c4.png">

**Reply by squidfunk** [reply]
You can't change the mount point to `${PWD}/src`. It needs to be `${PWD}/docs`, as it's the working directory within the Docker image: https://github.com/squidfunk/mkdocs-material/blob/73602b23e158c42021d41a31e254d8448f924aa6/Dockerfile#L70-L71
